infrastructure/datastore/db.go

```diff
@@ -0,0 +1,110 @@
+package datastore
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strconv"
+	"taskmanager/domain/model"
+)
+
+type dbType struct {
+	tasks []*model.Task
+}
+
+// DB interface contains the methods needed for query data
+type DB interface {
+	FindByID(id uint) (*model.Task, error)
+	FindAll() []*model.Task
+}
+
+// NewDB returns a new DB instance
+func NewDB() (DB, error) {
+	csvfile, err := os.Open("tasks.csv")
+	if err != nil {
+		log.Fatalln("Could not open the CSV file", err)
+	}
+	defer csvfile.Close()
+
+	tasks, err := populateTasks(csv.NewReader(csvfile))
+	if err != nil {
+		return nil, fmt.Errorf("Could not open DB")
+	}
+
+	populatedDB := &dbType{tasks: tasks}
+
+	return populatedDB, nil
+}
+
+func (d *dbType) FindByID(id uint) (*model.Task, error) {
+	for _, task := range d.tasks {
+		if task.ID == id {
+			return task, nil
+		}
+	}
+	return nil, fmt.Errorf("Could not find ID: %v", id)
+}
+
+func (d *dbType) FindAll() []*model.Task {
+	return d.tasks
+}
+
+func populateTasks(reader *csv.Reader) ([]*model.Task, error) {
+	tasks := []*model.Task{}
+
+	for {
+		record, err := reader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+
+		task, err := recordToTask(record)
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+
+		tasks = append(tasks, &task)
+	}
+
+	if len(tasks) < 0 {
```

Comment:
Can this happen? Should this validation be `len(tasks) == 0` instead?

Reply:
It cannot happen, I did not thing that condition correctly 😆 I change for `len(tasks) == 0` as you mention